src/Repository/AbstractRepository.php

```diff
@@ -344,35 +346,56 @@ public function getEntitiesFromTarget(EntityInterface $entity, EntityInterface $
     }
 
     /**
-     * @param array $filters
-     * @return mixed
+     * Returns entities that contain the $fieldValue of the $fieldName.
+     * The match is performed using the LIKE comparison operator.
+     * The result set is paginated.
+     *
+     * @param string $fieldName
+     * @param string $fieldValue
+     * @param int $from
+     * @param int $size
+     * @return array
      */
-    public function getNumberOfEntities(array $filters)
+    public function getEntitiesByField(string $fieldName = 'id', string $fieldValue ="", int $from = 0, int $size = 999999999999999)
     {
-        $query = "SELECT count(*) as entitiesNumber FROM " .
-            $this->getTableName() .
-            $this->getFilters($filters);
-        $dbStmt = $this->pdo->prepare($query);
-        foreach ($filters as $fieldName => &$value) {
-            $dbStmt->bindParam(':' . $fieldName, $value);
+        $query = 'SELECT count(*) as entitiesNumber FROM ' . $this->getTableName();
+        if ($fieldName) {
+            $query = 'SELECT * FROM ' . $this->getTableName() . ' WHERE ' . $fieldName . ' LIKE :field LIMIT :limit OFFSET :offset;';
         }
-        $dbStmt->execute();
-
-        return $dbStmt->fetch();
-    }
-
-    public function searchByField(string $field, string $text)
-    {
-        $query = "SELECT * FROM " . $this->getTableName() . " WHERE " . $field . " LIKE '%" . $text . "%'";
         $dbStmt = $this->pdo->prepare($query);
+        $dbStmt->bindParam(':limit', $size);
+        $dbStmt->bindParam(':offset', $from);
+        $dbStmt->bindValue(':field', "%$fieldValue%");
         $dbStmt->execute();
         $rows = $dbStmt->fetchAll();
         $result = [];
+
         foreach ($rows as $row) {
             $result[] = $this->hydrator->hydrate($this->entityName, $row);
         }
 
         return $result;
     }
 
+    /**
+     * Returns number of entities that contain the $fieldValue of the $fieldName.
+     * The match is performed using the LIKE comparison operator.
+     *
+     * @param string $fieldName
+     * @param string $fieldValue
+     * @return mixed
+     */
+    public function getEntityNumberByField(string $fieldName = null, string $fieldValue = "")
```

Comment:
return type :)

Reply:
also the naming is ambiguous. get Entity Number sounds like getting a specific entity number, not a number of entities